public/app/features/explore/utils/dateTimeProcessor.ts

```diff
@@ -0,0 +1,5 @@
+import { dateTime, DateTime, isDateTime } from '@grafana/data';
+
+export const processDateTime = (dt: DateTime | string) => {
+  return isDateTime(dt) ? dateTime(dt) : dt;
+};
```

Comment:
No, because we just make copies if they are moment objects (because they are mutable). Here you have a similar example in [dashboard/services/TimeSrv](https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/services/TimeSrv.ts#L280-L294)